src/main/scala/is/hail/expr/ir/functions/Functions.scala

```diff
@@ -7,38 +7,60 @@ import is.hail.utils._
 import is.hail.asm4s.coerce
 
 import scala.collection.mutable
-import scala.reflect.ClassTag
 
 object IRFunctionRegistry {
 
-  val registry: mutable.Map[String, Seq[(Seq[Type], Seq[IR] => IR)]] = mutable.Map().withDefaultValue(Seq.empty)
+  val irRegistry: mutable.Map[String, Seq[(Seq[Type], Seq[IR] => IR)]] = mutable.Map().withDefaultValue(Seq.empty)
+
+  val codeRegistry: mutable.Map[String, Seq[(Seq[Type], IRFunction)]] = mutable.Map().withDefaultValue(Seq.empty)
```

Comment:
@catoverdrive Sorry I'm late to the game here. We should use [`mutable.MultiMap`](https://www.scala-lang.org/api/2.12.3/scala/collection/mutable/MultiMap.html) when we have `Map[K, Set[V]]`. Unless there's a compelling case for `Seq` here?

Reply:
That also makes `addIRFunction` and `addIR` just calls to `irRegistry.addBinding` and `codeRegistry.addBinding`

Reply:
womp, I didn't know this was a thing. I'll change.